### ISSUE TITLE
[Woo POS] Fix clipping issues 

### ISSUE THREAD

**Issue by backwardstruck** [woocommerce/woocommerce-android#12575]
This issue is to address all clipping issues mentioned in P2 for POS. More context [here](https://wp.me/pdfdoF-5iv#comment-6389)

Design. Products Screen bottom This shadow should be clipped to the line. If this is too technically difficult, please just have the second transport dark line instead.

<img width="468" alt="Screenshot 2024-09-11 at 16 58 06" src="https://github.com/user-attachments/assets/3e2a7103-b720-4d06-af92-80700af7102d">

❓ Design. Products Screen bottom Adding a product causes the clipping to break briefly. This looks better when the other views are fixed.

<img width="407" alt="Screenshot 2024-09-11 at 16 58 36" src="https://github.com/user-attachments/assets/fe6cc3cd-9f4b-4f1b-92a7-5563a4120684">

❓ Design. Products Screen bottom This shadow also needs to be clipped at the top.

<img width="728" alt="Screenshot 2024-09-11 at 16 59 05" src="https://github.com/user-attachments/assets/472bd75e-1b13-48f7-8ec8-fbc4fb872c15">

❓ Design. Checkout Screen bottom Something is clipping the cards/shadows here before the edge of the screen.

<img width="794" alt="Screenshot 2024-09-11 at 16 59 36" src="https://github.com/user-attachments/assets/fd235b1b-7130-486e-a7cc-fe7f426b0232">

**Comment by dangermattic** [comment]
🚫 Please add a type label (e.g. **type: enhancement**) and a feature label (e.g. **feature: stats**) to this issue.

**Comment by backwardstruck** [comment]
![Image](https://github.com/user-attachments/assets/7b9688de-6edf-4e11-a800-428c842e2176)

**Comment by backwardstruck** [comment]
@joe-keenan as far as I can tell, this other issue listed is already fixed:

"❓ Design. Checkout Screen bottom Something is clipping the cards/shadows here before the edge of the screen."

Here's a screenshot of [my PR](https://github.com/woocommerce/woocommerce-android/pull/12588):

![Screenshot_20240917_162253](https://github.com/user-attachments/assets/ac3fdea6-289a-44e1-b305-6385a63b39f9)

**Comment by joe-keenan** [comment]
Hey @backwardstruck, the screenshot in your comment shows the products/cart screen, but my comment was referring to the cart/checkout screen. I think what’s happening is that the white background behind the “Check out” button doesn’t entirely disappear when the cart transitions.

**Comment by backwardstruck** [comment]
> Hey @backwardstruck, the screenshot in your comment shows the products/cart screen, but my comment was referring to the cart/checkout screen. I think what’s happening is that the white background behind the “Check out” button doesn’t entirely disappear when the cart transitions.

Sorry, you're right. I shouldn't have cropped that screenshot so closely. In that case this should probably be a separate ticket. Thanks for the additional info. To me it looks like we fixed that when we removed some of the Compose views that were nested there. Let me know if it's still not right.

Here's how trunk looks on the cart/checkout screen:

![Screenshot_20240923_153131](https://github.com/user-attachments/assets/f6d4a77a-15ca-465b-bfe6-6a134d610415)

![Screenshot_20240923_153322](https://github.com/user-attachments/assets/5f50d5fe-30dd-40f5-b83a-165fe374227e)
